src/meta/src/controller/fragment.rs

```diff
@@ -1105,7 +1105,19 @@ impl CatalogController {
             .into_tuple()
             .all(&inner.db)
             .await?;
-        Ok(actors)
+        Ok(actors
+            .into_iter()
+            .map(|(id, upstream_ids)| {
+                (
+                    id,
+                    upstream_ids
+                        .into_inner()
+                        .into_iter()
+                        .flat_map(|(_, ids)| ids.into_iter())
+                        .collect(),
```

Comment:
I think we can return the UpstreamActorIds as-is here, and only `map` once in the caller site